### PR TITLE
decompiling func_us_80174FD0

### PR DESCRIPTION
**Pull request by Onenutmcgee**
Nothing too crazy.  This is the 0xD2 update function.  Still unsure what it is.  It looks like it is the "seek enemy" mode when it rolls attack.  0xD3 will likely confirm or reject that theory.

Full match both
psp: https://decomp.me/scratch/yEZu5
psx: https://decomp.me/scratch/w055D

### CODE REVIEW

File: src/servant/tt_003/demon.c [hunk 1]
```diff
@@ -511,7 +522,102 @@ void UpdateServantDefault(Entity* self) {
...
+        self->step++;
+        // fallthrough
+    case 2:
+        if (!CheckEntityValid(self->ext.demon.target) &&
+            !(self->ext.demon.target = FindValidTarget(self))) {
+            self->entityId = DEMON_MODE_DEFAULT_UPDATE;
+            self->step = 0;
+        } else {
+            D_us_801785F0 = self->ext.demon.target->posX.val;
+            D_us_801785F0 += D_us_80178604 ? 0x200000 : -0x200000;
```
Comment (sozud) [inline src/servant/tt_003/demon.c:544]: These are FIX I think?

File: src/servant/tt_003/demon.c [hunk 2]
```diff
@@ -511,7 +522,102 @@ void UpdateServantDefault(Entity* self) {
...
+            self->velocityX = (D_us_801785F0 - self->posX.val) >> 3;
+            self->velocityY = (D_us_801785F4 - self->posY.val) >> 3;
+
+            self->facingLeft = self->velocityX > 0 ? 0 : 1;
+            self->posX.val += self->velocityX;
+
+            self->posY.val += self->velocityY;
+            D_us_801785F8 = abs(D_us_801785F0 - self->posX.val);
+            D_us_801785FC = abs(D_us_801785F4 - self->posY.val);
+            if ((D_us_801785F8 <= 0x7FFFF) && (D_us_801785FC <= 0xFFFF)) {
```
Comment (sozud) [inline src/servant/tt_003/demon.c:556]: What do these look like as FIX?

File: src/servant/tt_003/demon.c [hunk 3]
```diff
@@ -511,7 +522,102 @@ void UpdateServantDefault(Entity* self) {
...
+        self->step++;
+        break;
+    case 4:
+        if (D_us_80178600 == -2) {
+            self->ext.demon.abilityTimer = 0;
+            self->step++;
+        }
+        break;
+    case 5:
+        self->velocityX = self->facingLeft ? 0x2000 : -0x2000;
```
Comment (sozud) [inline src/servant/tt_003/demon.c:592]: FIX?

**Review by sozud — APPROVED** [review]

File: src/servant/tt_003/demon.c (hunk 1 above)
Comment (Onenutmcgee) [reply src/servant/tt_003/demon.c:544]: Oh yeah. I forgot about FIX. I’ll clean that up here tonight or tomorrow

**Pull request merged**